### ISSUE TITLE
Multiple spaces found before string

### ISSUE THREAD

**Issue by dangreen** [eslint/eslint#7487]
**Tell us about your environment**
- **ESLint Version:** 3.8.1
- **Node Version:** 6.3.0
- **npm Version:** 3.10.3

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint
**Please show your full configuration:**

```
module.exports = {
    "root": true,
    "parser": "babel-eslint",
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType":  "module",
        "ecmaFeatures": {
            "impliedStrict": true,
            "experimentalObjectRestSpread": true,
            "jsx": true
        }
    },
    "globals": {
        "__":            true,
        "__n":           true,
        "SITE_DATA":     true,
        "Stalk":         true,
        "HOST":          true,
        "PROVIDER_HOST": true,
        "SENTRY":        true,
        "LIFTED_AT":     true,
        "FORM_ACTION":   true,
        "ymaps":         true,
        "google":        true
    },
    "env": {
        "es6":      true,
        "browser":  true,
        "commonjs": true
    },
    "plugins": [
        "react"
    ],
    "extends": "eslint:recommended",
    "rules": {
        // Possible Errors
        "no-console":                   ["error", { allow: ["warn", "error"] }],
        // "valid-jsdoc":                   "warn",

        // Best Practices
        "array-callback-return":         "error",
        "block-scoped-var":              "error",
        // "complexity":                   ["error", 5],
        "consistent-return":             "error",
        "curly":                         "error",
        "default-case":                  "error",
        "dot-location":                 ["error", "property"],
        "dot-notation":                  "error",
        // "eqeqeq":                     "error",
        "no-caller":                     "error",
        "no-case-declarations":          "error",
        "no-eval":                       "error",
        "no-extra-label":                "error",
        "no-implicit-coercion":         ["error", { "allow": ["~"] }],
        "no-implicit-globals":           "error",
        "no-implied-eval":               "error",
        // "no-invalid-this":               "error",
        "no-iterator":                   "error",
        "no-lone-blocks":                "error",
        "no-loop-func":                  "error",
        // "no-magic-numbers":              "error",
        "no-multi-spaces":              ["error", { exceptions: { 
            "ImportDeclaration":  true,
            "VariableDeclarator": true,
            "Property":           true
        }}],
        "no-multi-str":                  "error",
        "no-native-reassign":            "error",
        "no-new-func":                   "error",
        "no-new-wrappers":               "error",
        "no-new":                        "error",
        "no-octal-escape":               "error",
        "no-param-reassign":            ["error", { "props": false }],
        "no-proto":                      "error",
        // "no-return-assign":              "error",
        "no-script-url":                 "error",
        "no-self-compare":               "error",
        "no-sequences":                  "error",
        "no-throw-literal":              "error",
        "no-unused-expressions":         "error",
        "no-useless-call":               "error",
        "no-useless-concat":             "error",
        "no-useless-escape":             "error",
        // "no-void":                       "error",
        "no-with":                       "error",
        "radix":                         "error",
        "yoda":                         ["error", "never", { "exceptRange": true }],

        // Variables
        "init-declarations":            ["error", "always"],
        "no-catch-shadow":               "error",
        "no-label-var":                  "error",
        "no-shadow-restricted-names":    "error",
        "no-undef-init":                 "error",
        "no-undefined":                  "error",
        "no-use-before-define":         ["error", { 
            "functions": false,
            "classes":   false
        }],

        // Node.js and CommonJS
        "callback-return":               "error",
        "handle-callback-err":           "error",
        "no-new-require":                "error",
        "no-path-concat":                "error",
        "no-process-env":                "error",
        "no-process-exit":               "error",
        "no-sync":                       "error", // let's see

        // Stylistic Issues
        "array-bracket-spacing":        ["error", "never"],
        "block-spacing":                 "error",
        "brace-style":                  ["error", "1tbs"],
        "camelcase":                     "warn", // let's see
        "comma-dangle":                 ["error", "never"],
        "comma-spacing":                 "error",
        "comma-style":                   "error",
        "computed-property-spacing":     "error",
        "consistent-this":              ["error", "self"],
        "eol-last":                      "error",
        "func-names":                    "error",
        "func-style":                   ["error", "declaration", { "allowArrowFunctions": true }],
        "indent":                       ["error", "tab", { "SwitchCase": 1 }],
        "jsx-quotes":                   ["error", "prefer-double"],
        "key-spacing":                  ["error", { "mode": "minimum", "align": "value" }],
        "keyword-spacing":               "error",
        "linebreak-style":               "error",
        "max-len":                      ["error", { 
            "code":                   140, 
            "tabWidth":               4,
            "ignoreTrailingComments": true, 
            "ignoreUrls":             true,
            "ignoreStrings":          true,
            "ignoreTemplateLiterals": true
        }],
        "max-nested-callbacks":         ["error", 4],
        "max-params":                   ["error", 6],
        "max-statements-per-line":       "error",
        "new-cap":                      ["error", { "capIsNew": false }],
        "new-parens":                    "error",
        "newline-after-var":             "error",
        // "newline-before-return":         "error",
        // "newline-per-chained-call":      "error",
        "no-array-constructor":          "error",
        "no-lonely-if":                  "error",
        "no-multiple-empty-lines":      ["error", { "max": 1 }],
        "no-new-object":                 "error",
        "no-spaced-func":                "error",
        "no-trailing-spaces":            "error",
        "no-unneeded-ternary":           "error",
        "no-whitespace-before-property": "error",
        "object-curly-spacing":         ["error", "always"],
        "one-var-declaration-per-line": ["error", "initializations"],
        "one-var":                      ["warn", "always"],
        "operator-assignment":          ["error", "always"],
        "operator-linebreak":           ["error", "before", { "overrides": { "=": "after" }}],
        "quote-props":                  ["error", "consistent"],
        "quotes":                       ["error", "single", { "avoidEscape": true, "allowTemplateLiterals": true }],
        "require-jsdoc":                 "warn",
        "semi-spacing":                  "error",
        "semi":                          "error",
        "space-before-blocks":           "error",
        "space-before-function-paren":  ["error", "never"],
        "space-in-parens":               "error",
        "space-infix-ops":              ["error", {"int32Hint": true}],
        "space-unary-ops":               "error",
        "spaced-comment":                "error",
        "unicode-bom":                   "error",

        // ECMAScript 6
        "arrow-body-style":             ["error", "as-needed"],
        "arrow-parens":                 ["error", "as-needed", { "requireForBlockBody": true }],
        "arrow-spacing":                 "error",
        "generator-star-spacing":       ["error", { "before": true, "after": false }],
        "no-confusing-arrow":           ["error", { "allowParens": true}],
        "no-duplicate-imports":          "error",
        "no-useless-computed-key":       "error",
        "no-useless-constructor":        "error",
        "no-useless-rename":             "error",
        "no-var":                        "error",
        "object-shorthand":             ["error", "always", { "avoidQuotes": true }],
        "prefer-arrow-callback":        ["error", { "allowNamedFunctions": true }],
        "prefer-const":                 ["error", {
            "destructuring":          "all",
            "ignoreReadBeforeAssign": false
        }],
        "prefer-reflect":                "error",
        "prefer-rest-params":            "error",
        "prefer-spread":                 "error",
        "prefer-template":               "error",
        "rest-spread-spacing":          ["error", "never"],
        // "sort-imports":                 ["error", {
        //     "ignoreCase": true,
        //     "ignoreMemberSort": true
        // }],
        "template-curly-spacing":        "error",
        "yield-star-spacing":           ["error", "before"],

        // React
        // TODO "react/forbid-prop-types", "react/prop-types", "react/require-optimization"
        "react/no-danger-with-children": "error",
        "react/no-deprecated":           "error",
        "react/no-is-mounted":           "error",
        "react/no-render-return-value":  "error",
        "react/no-string-refs":          "error",
        "react/prefer-es6-class":        "error",
        "react/react-in-jsx-scope":      "error",
        "react/require-render-return":   "error",
        "react/self-closing-comp":       "error",
        "react/sort-comp":              ["warn", { order: [
            "static-methods",
            "type-annotations",
            "render",
            "lifecycle",
            "everything-else"
        ]}],
        "react/style-prop-object":            "error",
        "react/jsx-boolean-value":           ["error", "never"],
        "react/jsx-closing-bracket-location": "warn",
        "react/jsx-curly-spacing":           ["error", "never"],
        "react/jsx-first-prop-new-line":     ["error", "multiline"],
        // "react/jsx-indent":                  ["error", "tab"],
        "react/jsx-indent-props":            ["error", "tab"],
        "react/jsx-key":                      "error",
        "react/jsx-max-props-per-line":      ["error", { "maximum": 3 }],
        // "react/jsx-no-bind":                 ["error", {
        //  "ignoreRefs": true,
        //  "allowBind":  true
        // }],
        "react/jsx-no-comment-textnodes":     "error",
        "react/jsx-no-duplicate-props":       "error",
        "react/jsx-no-undef":                 "error",
        "react/jsx-pascal-case":              "error",
        "react/jsx-space-before-closing":    ["error", "never"],
        "react/jsx-uses-react":               "error",
        "react/jsx-uses-vars":                "error",
        "react/jsx-wrap-multilines":          "error"
    }
};
```

**What did you do? Please include the actual source code causing the issue.**

``` js
import LandingComponent, { getComponents, bound_actions, extra_bound_actions, style } from '../landing-component';
import { config, listener } from 'lib/base-component';
import React, { PropTypes } from 'react';

import SectionStyle from './section.sasst';
import Configure    from './section-configure';

@config(Configure)
@extra_bound_actions({
    classes: 'icon icon-switch',
    event:   'changeTemplate'
})
@bound_actions({ createBefore: false, createAfter: false })
@style(SectionStyle)
export default class Section extends LandingComponent {

...
```

**What did you expect to happen?**
no errors
**What actually happened? Please include the actual, raw output from ESLint.**
`11:11  error    Multiple spaces found before ''changeTemplate''        no-multi-spaces`

**Comment by not-an-aardvark** [comment]
Thanks for the report. Are you able to reproduce this without `babel-eslint`? I'm wondering whether the decorator syntax is causing the problem.

**Comment by dangreen** [comment]
@not-an-aardvark I think only babel can parse decorators.

**Comment by not-an-aardvark** [comment]
To clarify, I was asking if you were able to reproduce it by removing both `babel-eslint` and the decorators (to see if the bug also affects code without experimental features).

**Comment by dangreen** [comment]
@not-an-aardvark without decorators all works perfectly.

**Comment by not-an-aardvark** [comment]
Cool, thanks for checking that.

However, at the moment ESLint generally doesn't support experimental features in core rules), so I don't think we will be changing the core rule to handle things like decorators. You might want to file an issue on [eslint-plugin-babel](https://github.com/babel/eslint-plugin-babel); that repo has some patched rules to support experimental features.

**Comment by vitorbal** [comment]
Closing, as the issue seems to be resolved. Please reopen if there is more to discuss.
